presto-parquet/src/main/java/com/facebook/presto/parquet/batchreader/decoders/Decoders.java

```diff
@@ -173,11 +234,46 @@ private static final ValuesDecoder createValuesDecoder(ColumnDescriptor columnDe
 
         if ((encoding == DELTA_BYTE_ARRAY || encoding == DELTA_LENGTH_BYTE_ARRAY) && type == PrimitiveTypeName.BINARY) {
             ByteBufferInputStream inputStream = ByteBufferInputStream.wrap(ByteBuffer.wrap(buffer, offset, length));
+
+            Optional<Type> prestoType = createDecimalType(columnDescriptor);
```

Comment:
I know this is from existing code, but it is confusing to detect if the logical type is decimal by calling createDecimalType() and check its returned type. It'll be clearer to make the type detecting consistent with other encodings, e.g. as what you did on line 195, 196

Reply:
Already refactored, thanks.